File: contests/abc199/abc199_b.py
n=int(input())
a=[*map(int,input().split())]
b=[*map(int,input().split())]
l=0
u=1001
for i,j in zip(a,b):
    l=max(i,l)
    u=min(j,u)
print(max(0,u-l+1))

File: rust/others/src/bin/abc199_b.rs
fn main() {
    proconio::input! {
        n: isize,
        a: [isize; n],
        b: [isize; n]
    }
    let mut u = 1001;
    let mut l = 0;
    for (i, j) in a.iter().zip(b.iter()) {
        if *i > l {
            l = *i
        }
        if *j < u {
            u = *j
        }
    }
    let c = u - l + 1;
    println!("{}", if c < 0 { 0 } else { c });
}
